metrics/api/src/main/java/io/helidon/metrics/api/Metadata.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.metrics.api;
+
+/**
+ * Holder for metadata to declare meters, compatible with that from Microprofile, to ease the transition to the current
+ * Helidon metrics API.
+ *
+ * @param name name for the meter
+ * @param description description for the meter
+ * @param unit unit for the meter
+ *
+ * @deprecated Use the various meter builders to prepare their characteristics, for example
+ * {@link io.helidon.metrics.api.Gauge.Builder}
+ */
+@Deprecated(since = "4.0.0")
+public record Metadata(String name, String description, String unit) {
```

Comment:
Metadata is a very general name. Can we rename to `MetricMetadata` (if that is what this is)

Reply:
Also using a record is reducing our options for maintenance - we can never change to interface, as this always has a public constructor. I would prefer to change to class or interface.

Reply:
Also according to the methods, we seem to be expecting that some of the fields may be null - in such a case this should never be our public API

Reply:
`Metadata` is here only to simplify migration from the previous metrics API. Existing code (in Helidon and, more importantly, in customer's projects) used the MP metrics `Metadata` to prepare SE metrics.

Adding this would allow that code to continue to build and run with a change in the `import` statement for it in customer's code.

I know 4.0 is a major version change and now is the time to break APIs, and that if we introduce our own `Metadata` then we must live with it until 5.0. Probably _not_ adding our own `Metadata` for migration is the better choice and get our code and users' code converted entirely to the new API.

Reply:
I removed `Metadata`. (There is one place where something like that was still useful in one component to avoid repeating the same logic many times but it is private to that component.)